ocaml/testsuite/tests/typing-layouts-bits32/basics.ml

```diff
@@ -484,29 +485,33 @@ Error: The native code version of the primitive is mandatory
 
 external f10_6 : (int32#[@unboxed]) -> bool -> string  = "foo" "bar";;
 [%%expect{|
-Line 1, characters 18-24:
-1 | external f10_6 : (int32#[@unboxed]) -> bool -> string  = "foo" "bar";;
-                      ^^^^^^
-Error: Don't know how to unbox this type.
-       Only float, int32, int64, nativeint, and vector primitives can be unboxed.
+external f10_6 : int32# -> bool -> string = "foo" "bar"
```

Comment:
I have a sinking feeling that dropping the attribute in the pretty-printer will come back to bite us. I can't tell you where, but it's just a hunch.

Reply:
I updated it to print the `[@unboxed]` attributes. It has the slightly unfortunate consequence of printing the attribute on all non-value types even when it's not annotated:

```
external f : float# -> float# = "%identity";;
[%%expect{|
external f : float# -> float# = "%identity" [@@unboxed]
|}]
```

Think this is mostly fine and better than not printing the attribute when explicitly annotated.

This issue is limited to `Printtyp` and does not affect `Pprintast` (used for `-dsource`). To fix it, we need to add a new `Unboxed_same_as_ocaml_repr` constructor which doesn't seem to be worth it.

Reply:
As discussed, let's print only when `-only_erasable_extensions` is set.

Reply:
Made the change in the latest commit